### PR TITLE
Setup Classes

### PR DESCRIPTION
**Pull request by Othniel2471**
### In this PR I:

- Created class Person

> Added Instance vars: `@id`, `@name`, and `@age`.
> Added Constructor with `name`, `age`, and `parent_permission` as parameters. Where `name `and `parent_permission` have and have default values of "Unknown" and true.
> Added Getters for `@id`, `@name`, and `@age`.
> Added Setters for `@name` and `@age`.
> Added Private method of_age? that returns true if `@age` is greater or equal to 18 and false otherwise.
> Added Public method can_use_services? that returns true if a person is of age or if they have permission from parents.

- Created class Student.
> Inherits from Person.
> Added a Constructor that extends the parent's constructor by adding `@classroom` and a parameter for it.
> Added Method play_hooky that returns "¯\(ツ)/¯".

- Created class Teacher.
> Inherits from Person.
> Added Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
> Override can_use_services? so it always returns true.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age
+    @age >= 18
+  end
+
+  def can_use_services?
+    @parent_permission || of_age?
+  end
```
Comment (Ibizugbe) [inline person.rb]: Nice job implementing the `of_age` method as a private method👍🏼. But kindly note that the `can_use_service` is meant to be a public method😉. I suggest you take the `can_use_service` method to go above the `private` declaration.

```suggestion

  def can_use_services?
    @parent_permission || of_age?
  end
  
  private

  def of_age
    @age >= 18
  end

```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Othniel2471 👋🏼👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] an individual .rb file for each class (Person, Student, and Teacher ) has been created👍🏼.
- [x] `getters` and `setters` for name and age has been created in the person class👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Othniel2471) [reply person.rb]: okay @Ibizugbe, I'll do that now, thank you

**Review by Ibizugbe — APPROVED** [review]
Hi @Othniel2471 👋🏼👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] the `can_use_method` has been made public as required👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**